Babel 12.2.x test failures due to no-breaking spaces

Hello, in Fedora we are updating our babel to 12.2.0 and I noticed some test failures of flask-babel. To reproduce:

```
[flask-babel (master *)]$ git diff
diff --git a/pyproject.toml b/pyproject.toml
index 9538cb9..190cc91 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -32,7 +32,7 @@ include = [
 python = "^3.7"
 pytz = "^2022.7"
 Flask = "^2.0.0"
-Babel = "^2.11.0"
+Babel = "^2.12.0"
 Jinja2 = "^3.1.2"

[flask-babel (master *)]$ poetry install
...
  • Installing babel (2.12.1)
...

[flask-babel (master *)]$ poetry run pytest
============================= test session starts ==============================
platform linux -- Python 3.11.2, pytest-7.2.1, pluggy-1.0.0
rootdir: .../flask-babel
plugins: cov-3.0.0, mock-3.10.0
collected 29 items

tests/test_date_formatting.py F.FF                                       [ 13%]
tests/test_force_locale.py ....                                          [ 27%]
tests/test_gettext.py ............                                       [ 68%]
tests/test_integration.py .......                                        [ 93%]
tests/test_multiple_apps.py .                                            [ 96%]
tests/test_number_formatting.py .                                        [100%]

=================================== FAILURES ===================================
_________________________________ test_basics __________________________________

    def test_basics():
        app = flask.Flask(__name__)
        babel.Babel(app)
        d = datetime(2010, 4, 12, 13, 46)
        delta = timedelta(days=6)
    
        with app.test_request_context():
>           assert babel.format_datetime(d) == 'Apr 12, 2010, 1:46:00 PM'
E           AssertionError: assert 'Apr 12, 2010...46:00\u202fPM' == 'Apr 12, 2010, 1:46:00 PM'
E             - Apr 12, 2010, 1:46:00 PM
E             ?                      ^
E             + Apr 12, 2010, 1:46:00 PM
E             ?                      ^

tests/test_date_formatting.py:16: AssertionError
_________________________ test_custom_locale_selector __________________________

    def test_custom_locale_selector():
        app = flask.Flask(__name__)
        b = babel.Babel(app)
        d = datetime(2010, 4, 12, 13, 46)
    
        the_timezone = 'UTC'
        the_locale = 'en_US'
    
        def select_locale():
            return the_locale
    
        def select_timezone():
            return the_timezone
    
        get_babel(app).locale_selector = select_locale
        get_babel(app).timezone_selector = select_timezone
    
        with app.test_request_context():
>           assert babel.format_datetime(d) == 'Apr 12, 2010, 1:46:00 PM'
E           AssertionError: assert 'Apr 12, 2010...46:00\u202fPM' == 'Apr 12, 2010, 1:46:00 PM'
E             - Apr 12, 2010, 1:46:00 PM
E             ?                      ^
E             + Apr 12, 2010, 1:46:00 PM
E             ?                      ^

tests/test_date_formatting.py:67: AssertionError
_______________________________ test_refreshing ________________________________

    def test_refreshing():
        app = flask.Flask(__name__)
        babel.Babel(app)
        d = datetime(2010, 4, 12, 13, 46)
        with app.test_request_context():
>           assert babel.format_datetime(d) == 'Apr 12, 2010, 1:46:00 PM'
E           AssertionError: assert 'Apr 12, 2010...46:00\u202fPM' == 'Apr 12, 2010, 1:46:00 PM'
E             - Apr 12, 2010, 1:46:00 PM
E             ?                      ^
E             + Apr 12, 2010, 1:46:00 PM
E             ?                      ^

tests/test_date_formatting.py:81: AssertionError
=========================== short test summary info ============================
FAILED tests/test_date_formatting.py::test_basics - AssertionError: assert 'A...
FAILED tests/test_date_formatting.py::test_custom_locale_selector - Assertion...
FAILED tests/test_date_formatting.py::test_refreshing - AssertionError: asser...
========================= 3 failed, 26 passed in 0.16s =========================
```

This looks like a change from a SPACE to NARROW NO-BREAK SPACE in babel output.

This is just a friendly heads-up for you, I don't expect you to do anything.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -32,7 +32,7 @@ include = [
 python = "^3.7"
 pytz = "^2022.7"
 Flask = "^2.0.0"
-Babel = "^2.11.0"
+Babel = "^2.12.0"
 Jinja2 = "^3.1.2"
 
 [tool.poetry.dev-dependencies]

--- a/tests/test_date_formatting.py
+++ b/tests/test_date_formatting.py
@@ -13,22 +13,22 @@ def test_basics():
     delta = timedelta(days=6)
 
     with app.test_request_context():
-        assert babel.format_datetime(d) == 'Apr 12, 2010, 1:46:00 PM'
+        assert babel.format_datetime(d) == 'Apr 12, 2010, 1:46:00\u202fPM'
         assert babel.format_date(d) == 'Apr 12, 2010'
-        assert babel.format_time(d) == '1:46:00 PM'
+        assert babel.format_time(d) == '1:46:00\u202fPM'
         assert babel.format_timedelta(delta) == '1 week'
         assert babel.format_timedelta(delta, threshold=1) == '6 days'
 
     with app.test_request_context():
         get_babel(app).default_timezone = 'Europe/Vienna'
-        assert babel.format_datetime(d) == 'Apr 12, 2010, 3:46:00 PM'
+        assert babel.format_datetime(d) == 'Apr 12, 2010, 3:46:00\u202fPM'
         assert babel.format_date(d) == 'Apr 12, 2010'
-        assert babel.format_time(d) == '3:46:00 PM'
+        assert babel.format_time(d) == '3:46:00\u202fPM'
 
     with app.test_request_context():
         get_babel(app).default_locale = 'de_DE'
         assert babel.format_datetime(d, 'long') == \
-            '12. April 2010 um 15:46:00 MESZ'
+            '12. April 2010, 15:46:00 MESZ'
 
 
 def test_custom_formats():
@@ -64,7 +64,7 @@ def test_custom_locale_selector():
     get_babel(app).timezone_selector = select_timezone
 
     with app.test_request_context():
-        assert babel.format_datetime(d) == 'Apr 12, 2010, 1:46:00 PM'
+        assert babel.format_datetime(d) == 'Apr 12, 2010, 1:46:00\u202fPM'
 
     the_locale = 'de_DE'
     the_timezone = 'Europe/Vienna'
@@ -78,7 +78,7 @@ def test_refreshing():
     babel.Babel(app)
     d = datetime(2010, 4, 12, 13, 46)
     with app.test_request_context():
-        assert babel.format_datetime(d) == 'Apr 12, 2010, 1:46:00 PM'
+        assert babel.format_datetime(d) == 'Apr 12, 2010, 1:46:00\u202fPM'
         get_babel(app).default_timezone = 'Europe/Vienna'
         babel.refresh()
-        assert babel.format_datetime(d) == 'Apr 12, 2010, 3:46:00 PM'
+        assert babel.format_datetime(d) == 'Apr 12, 2010, 3:46:00\u202fPM'

